### PR TITLE
Rewrite starter log message

### LINKED ISSUE #2212: A user could influence the syslog strings by symlinking the starter-suid
## Version of Singularity:

3.0.0 (running on Ubuntu 18.04.1)

### Expected behavior

The user should not be able to influence the log message in /var/log/syslog in this way.

### Steps to reproduce behavior

As a normal user run

    ubuntu@laptop:~$ export SINGULARITY_MESSAGELEVEL=1
    ubuntu@laptop:~$ ln -s /usr/local/libexec/singularity/bin/starter-suid foobar
    ubuntu@laptop:~$ ./foobar 
    ERROR  : SRUNTIME environment variable isn't set
    ubuntu@laptop:~$ 

As root run this command

    root@laptop:~# grep foobar /var/log/syslog
    Oct 10 14:09:27 laptop Singularity: foobar (U=0,P=17206)> SRUNTIME environment variable isn't set
    root@laptop:~# 

The relevant source code seems to be:

    ubuntu@laptop:~/go/src/github.com/sylabs/singularity$ grep syslog_st ./src/runtime/c/lib/util/message.c |grep snprintf
            if (snprintf(syslog_string, 540, "%s (U=%d,P=%d)> %s", __progname, geteuid(), getpid(), message) >= 540) {
    ubuntu@laptop:~/go/src/github.com/sylabs/singularity$ 

Note, Unix filenames are not just limited to alphanumeric characters. If I remember correctly everything but the character slash is permitted.

### PR DESCRIPTION
**Pull request by cclerget**
**Description of the Pull Request (PR):**

This PR rewrites C starter log message code, address issue #2212 and finally align debug output with Go sylog package.

**This fixes or addresses the following GitHub issues:**

- Fixes #2212 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](../CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](../CHANGELOG.md) if necessary according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](../CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: src/runtime/c/lib/util/message.c [hunk 1]
```diff
@@ -1,84 +1,74 @@
...
- * NOTICE.  This Software was developed under funding from the U.S. Department of
- * Energy and the U.S. Government consequently retains certain rights. As such,
- * the U.S. Government has been granted for itself and others acting on its
- * behalf a paid-up, nonexclusive, irrevocable, worldwide license in the Software
- * to reproduce, distribute copies to the public, prepare derivative works, and
- * perform publicly and display publicly, and to permit other to do so. 
- * 
+ * This software is licensed under a 3-clause BSD license. Please
+ * consult LICENSE.md file distributed with the sources of this
+ * project regarding your rights to use or distribute this software.
```
Comment (bauerm97) [inline src/runtime/c/lib/util/message.c]: We can't remove the `Regents of the University of California...` section of the copyright. This file is coming from the previous version which was developed with that copyright header so it must remain.

File: src/runtime/c/lib/util/message.h [hunk 2]
```diff
@@ -1,63 +1,43 @@
...
- * NOTICE.  This Software was developed under funding from the U.S. Department of
- * Energy and the U.S. Government consequently retains certain rights. As such,
- * the U.S. Government has been granted for itself and others acting on its
- * behalf a paid-up, nonexclusive, irrevocable, worldwide license in the Software
- * to reproduce, distribute copies to the public, prepare derivative works, and
- * perform publicly and display publicly, and to permit other to do so. 
- * 
+ * This software is licensed under a 3-clause BSD license. Please
+ * consult LICENSE.md file distributed with the sources of this
+ * project regarding your rights to use or distribute this software.
```
Comment (bauerm97) [inline src/runtime/c/lib/util/message.h]: Same comment about copyright here.

**Review by bauerm97 — CHANGES_REQUESTED** [review]

File: src/runtime/c/lib/util/message.c [hunk 3]
```diff
@@ -1,84 +1,86 @@
-/* 
- * Copyright (c) 2017-2018, SyLabs, Inc. All rights reserved.
+/*
+ * Copyright (c) 2018, SyLabs, Inc. All rights reserved.
  * Copyright (c) 2017, SingularityWare, LLC. All rights reserved.
```
Comment (bauerm97) [inline src/runtime/c/lib/util/message.c]: I think (@gmkurtzer or @jmstover correct me if I'm wrong) that we no longer need to include the SingularityWare copyright line, as when Sylabs was created we transferred all IP from SingularityWare -> Sylabs. So we should remove the SingularityWare line, and update the Sylabs line to be:

`Copyright (c) 2017-2018, Sylabs Inc. All rights reserved.`

**Review by bauerm97 — DISMISSED** [review]
I think after one small change to the copyright header this is good to go!

**Comment by cclerget** [comment]
@bauerm97 Fixed, thanks !

**Review by yhcote — APPROVED** [review]
LGTM

**Pull request merged**